### ISSUE TITLE
/var/run issues

### ISSUE THREAD

**Issue by tsumaru720** [opnsense/core#1485]
Hi 

Running OPNsense 17.1.3-amd64

it seems /var/run is cleared on boot which is a problem when running something like quagga.

Quagga expects /var/run/quagga to exist and be owned by quagga user - It doesn't seem to be retained on reboot thus preventing quagga from starting up again - In older versions of opnsense (16.x) this was not a problem. 

Is it possible to create a new folder here on boot as a specific user?

**Comment by fichtner** [comment]
Hi Scott,

Are you using /var RAM disk option (e.g. from Nano image)?


Cheers,
Franco

**Comment by fabianfrz** [comment]
@fichtner add the creation of this directory to the plugin?

**Comment by tsumaru720** [comment]
@fichtner honestly not sure - i used the ISO installer so ...

**Comment by fabianfrz** [comment]
@tsumaru720 this setting is under system advanced but you may be faster if you would enter `mount` into a shell and look for a `/var` mount.

**Comment by tsumaru720** [comment]
@fabianfrz 

Well, this is interesting because I do not have the option enabled :)

```
root@vpn1:/var/run # mount
/dev/gpt/rootfs on / (ufs, local, noatime, soft-updates)
devfs on /dev (devfs, local, multilabel)
tmpfs on /tmp (tmpfs, local)
tmpfs on /var (tmpfs, local)
devfs on /var/dhcpd/dev (devfs, local, multilabel)
```
![ff](https://cloud.githubusercontent.com/assets/239857/24083635/a25543c0-0cd2-11e7-874c-828eccb80f20.png)

Bug maybe?

**Comment by AdSchellevis** [comment]
@tsumaru720 is your config setup from scratch or imported?

**Comment by tsumaru720** [comment]
Imported (It wasn't a ram-disk their either)

**Comment by fabianfrz** [comment]
@tsumaru720 what does `cat /etc/fstab` print?

**Comment by AdSchellevis** [comment]
from pfSense? in that case it might be a different interpretation (configs aren't 100% compatible), just save again and reboot to fix.

**Comment by tsumaru720** [comment]
@fabianfrz 
```
root@vpn1:/var/run # cat /etc/fstab 
# Device                Mountpoint      FStype  Options         Dump    Pass#
/dev/gpt/rootfs /               ufs     rw,noatime      1       1
root@vpn1:/var/run #
```

@AdSchellevis Nope, was imported from OPNsense 16.7.14 :)

**Comment by tsumaru720** [comment]
Also ticked /tmp RAM Disk and saved / rebooted - /var is still a ram disk

**Comment by fabianfrz** [comment]
@tsumaru720 unchecked is correct - I am having `/tmp` a ram disk but not `/var` because of other issues and it works on my device as expected. I am also using `17.1.3`. So your device has a strange behaviour.

**Comment by fichtner** [comment]
Let's run this...

    # grep -c 'use_mfs_.*var' /conf/config.xml

**Comment by fichtner** [comment]
(small correction, no -c)

**Comment by tsumaru720** [comment]
```
root@vpn1:/usr/local/etc # grep -c 'use_mfs_.*var' /conf/config.xml
1
root@vpn1:/usr/local/etc # grep 'use_mfs_.*var' /conf/config.xml
    <use_mfs_var_size/>
```

**Comment by fichtner** [comment]
Oh right, that hasn't been used for a long time.

Can you edit config.xml and just remove this?

**Comment by fichtner** [comment]
After reboot, just reapply quagga package, the directory should be there:

    # opnsense-revert quagga

@fabianfrz we should add this to the plugin, but details on why/how follow

**Comment by fichtner** [comment]
For old time's sake... removed in 2015 :) 4b9a265f84e

**Comment by tsumaru720** [comment]
@fichtner that does appear to have fixed the problem :)

1. Reboot
2. re-create folder & set permissions
3. Start quagga
4. Reboot
5. still there :)

I assume it's safe to remove use_mfs_tmp_size too?

```
root@vpn1:/var/run # grep use_mfs /conf/config.xml 
    <use_mfs_tmp_size/>
    <use_mfs_tmp>1</use_mfs_tmp>
```
I wonder how it has survived in my config for so long? :D 
I did originally install 15.7 & have upgraded through to 16.7.14. Tried to do 17.1 but ran into disk space issues so it broke & had to reinstall (The issues are my own fault)

**Comment by fichtner** [comment]
Hi Simon,

Yes, use_mfs_tmp_size can be removed too.

I introduced this problem while splitting "/tmp /var MFS" into two separate options while trying to retain backwards compatibility. To be honest, one of the issues here is that we read XML via grep and that is bound to blow up eventually, but it's so early in the system boot that we can't just fire up the whole PHP... I'll issue a fix one way or another for 17.1.4. Thanks for spotting this. :)


Cheers,
Franco

**Comment by tsumaru720** [comment]
No problem at all - Keep up the good work with OPNsense! :)
